app/src/main/java/org/phenoapps/verify/LoaderDBActivity.java

```diff
@@ -532,32 +537,69 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
     }
 
+    public static String getPath(final Context context, final Uri uri){
+        String absolutePath = getLocalPath(context, uri);
+        return absolutePath != null ? absolutePath : uri.toString();
+    }
+
 
     //based on https://github.com/iPaulPro/aFileChooser/blob/master/aFileChooser/src/com/ipaulpro/afilechooser/utils/FileUtils.java
-    public String getPath(Uri uri) {
+    public static String getLocalPath(final Context context , Uri uri) {
```

Comment:
Is there a reason for making these static? Encapsulate uri handling / authority parsing into a separate class.